src/ORM/DataObject.php

```diff
@@ -940,6 +949,50 @@ public function i18n_plural_name()
         return _t(static::class . '.PLURALNAME', $this->plural_name());
     }
 
+    /**
+     * Get description for this class
+     * @return null|string
+     */
+    public function classDescription()
+    {
+        return static::config()->get('class_description', Config::UNINHERITED);
+    }
+
+    /**
+     * Get localised description for this class
+     * @return null|string
+     */
+    public function i18n_classDescription()
+    {
+        $placeholder = 'PLACEHOLDER_DESCRIPTION';
+        $baseDescription = $this->classDescription() ?? $placeholder;
+
+        // Check the new i18n key first
+        $description = _t(static::class . '.CLASS_DESCRIPTION', $baseDescription);
+        if ($description !== $baseDescription) {
+            return $description;
+        }
+
+        // Fall back on the deprecated localisation key
+        $legacyI18n = _t(static::class . '.DESCRIPTION', $baseDescription);
+        if ($legacyI18n !== $baseDescription) {
+            Deprecation::notice(
+                '5.4.0',
+                'The ' . static::class . '.DESCRIPTION' . ' i18n localisation key is deprecated. Use '
+                    . static::class . '.CLASS_DESCRIPTION' . ' instead.',
+                Deprecation::SCOPE_GLOBAL
+            );
+            return $legacyI18n;
+        }
```

Comment:
BC protection for custom classes where devs have added localisations for the description. Will be removed in 6.